src/ir/proto/sql/decoder_context.h

```diff
@@ -0,0 +1,93 @@
+//-----------------------------------------------------------------------------
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//----------------------------------------------------------------------------
+
+#ifndef SRC_IR_PROTO_SQL_MVP_DECODER_CONTEXT_H_
+#define SRC_IR_PROTO_SQL_MVP_DECODER_CONTEXT_H_
+
+#include <memory>
+#include <string>
+
+#include "absl/container/flat_hash_map.h"
+#include "src/ir/types/type_factory.h"
+#include "src/ir/ir_context.h"
+#include "src/ir/block_builder.h"
+#include "src/ir/storage.h"
+#include "src/ir/value.h"
+
+namespace raksha::ir::proto::sql {
+
+// Context and helper methods on context state for decoding SQL MVP proto
+// objects.
+class DecoderContext {
+ public:
+  static constexpr absl::string_view kTagClaimOperatorName = "tag_claim";
+  static constexpr absl::string_view kAddedTagAttributeName = "added_tag";
+  static constexpr absl::string_view kTagKindAttributeName = "tag_kind";
+  static constexpr absl::string_view kConfidentialityTagAttrValue =
+      "confidentiality";
+  static constexpr absl::string_view kIntegrityTagAttrValue = "integrity";
+  static constexpr absl::string_view kTagClaimInputValueName = "input";
+  static constexpr absl::string_view kTagClaimOutputName = "out";
+
+  explicit DecoderContext(IRContext &ir_context) : ir_context_(ir_context) { }
```

Comment:
Why do you need this class at all? Simply use `IRContext` directly?

Reply:
I was trying to put context-y type things that we would only use for the SQL MVP in here to not pollute the `IRContext` class. I can try to slim this down/eliminate it though.